### PR TITLE
Allow initializing FIRLocalCacheSettings with unlimited size.

### PR DESCRIPTION
**Pull request by tom-andersen**

### CODE REVIEW

File: Firestore/Source/API/FIRLocalCacheSettings.mm [hunk 1]
```diff
@@ -73,7 +73,8 @@ - (instancetype)init {
 
 - (instancetype)initWithSizeBytes:(NSNumber *)size {
   self = [super init];
-  if (size.longLongValue < Settings::MinimumCacheSizeBytes) {
+  if (size.longLongValue != kFIRFirestoreCacheSizeUnlimited &&
```
Comment (wu-hui) [inline Firestore/Source/API/FIRLocalCacheSettings.mm:76]: I thought we were going to use int64_t::max or something like that?

Also, please add a test to set kFIRFirestoreCacheSizeUnlimited

Comment (tom-andersen) [reply Firestore/Source/API/FIRLocalCacheSettings.mm]: As discussion today, using `-1` is more inline with other SDKs.

Test added.

File: Firestore/Example/Tests/Integration/API/FIRDatabaseTests.mm [hunk 2]
```diff
@@ -1813,4 +1815,10 @@ - (void)testMinimumCacheSize {
                           NSException);
 }
 
+- (void)testUnlimitedCacheSize {
+  FIRPersistentCacheSettings *settings =
+      [[FIRPersistentCacheSettings alloc] initWithSizeBytes:@(Settings::CacheSizeUnlimited)];
+  XCTAssertEqual(settings.internalSettings.size_bytes(), Settings::CacheSizeUnlimited);
```
Comment (wu-hui) [inline Firestore/Example/Tests/Integration/API/FIRDatabaseTests.mm:1821]: Can we try actually initialize the SDK with this settings object and do something (anything)?

**Review by wu-hui — APPROVED** [review]
LGTM with the condition to expand to test a bit more.

**Pull request merged**